src/app.rs

```diff
@@ -175,6 +177,42 @@ pub struct Config {
     pub using_table: Option<String>,
     pub using_port: Option<u32>,
     // pub cache_expiration_time: Option<i64>, // in second. default 300 (= 5 minutes)
+    #[serde(default)]
+    pub retry_batch_write_item: RetryConfig,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone, Default)]
+pub struct RetryConfig {
+    pub enabled: bool,
+    #[serde(default)]
+    pub jitter: bool,
+    pub factor: Option<f32>,
+    pub min_delay: Option<u64>,
```

Comment:
We must accept it as `f64` or `decimal` to specify the sub-second duration. From a semantic perspective, decimal is preferable because it does not lose information.

Reply:
I changed the type of `initial_delay` and `max_delay`  to `Duration` same as [aws-sdk](https://github.com/awslabs/aws-sdk-rust/blob/631b5f0af67964bbac4003db6a1e62a8a8597360/sdk/aws-smithy-types/src/retry.rs#L290C18-L290C26).
```
retry:
  default:
    initial_backoff: ~
    max_backoff:
      secs: 10
      nanos: 0
```

Reply:
The current implementation is an excellent first point to start. Thank you for pointing out `Duration`. I did not realize that it is serializable/deserializable.

As mentioned in [serde's PR](https://github.com/serde-rs/serde/issues/339#issuecomment-539453327), this is not a user-friendly format for a configuration file. For example, when you want to set `max_backoff` to `500` milliseconds, you need to configure it as follows:

```yaml
retry:
  default:
    initial_backoff:
      secs: 0
      nanos: 500000000
```

Do you happen to have any idea how to fix this problem? I believe it is challenging due to serialization architecture, but I would like to hear about it if you have any idea how to do it. For now, I am considering using `serde(untagged)`. However, I noticed that it can be addressed later if we use this strategy. If you are not interested in this improvement, you have an option to leave this to be addressed later. Please let me know your choice.
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=03d40820fa36919292aa0377e879a0b6

Reply:
> you have an option to leave this to be addressed later

This will take some time to respond, so please do this.

Reply:
I have created a new issue to track this improvement. https://github.com/awslabs/dynein/issues/225